vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/steps/integration/ResourceCheckSteps.java

```diff
@@ -216,9 +217,20 @@ private String getCssSelector(Element element)
     private URI resolveUri(String uri) throws URISyntaxException
     {
         URI uriToCheck = new URI(uri);
+
         if (isNotAbsolute(uriToCheck))
         {
             URI mainApplicationPageUrl = webApplicationConfiguration.getMainApplicationPageUrlUnsafely();
+            if (uri.charAt(1) == '/')
+            {
+                String scheme = Optional.ofNullable(mainApplicationPageUrl)
+                                        .map(URI::getScheme)
+                                        .orElseThrow(() -> new IllegalStateException(
+                                                "Unable to evaluate protocol for main application page url: "
+                                                        + mainApplicationPageUrl));
+                return new URIBuilder(uri).setScheme(scheme).build();
+            }
+
             if (mainApplicationPageUrl != null)
```

Comment:
this check is duplicated: here and `Optional.ofNullable`

Reply:
but we will never get to this point if we got to that check so I think its not critical